js/views/search/Search.js

```diff
@@ -0,0 +1,248 @@
+      this.sProvider = provider;
+      this.processTerm(this.term);
+    }
+  });
+  }
+
+  className() {
+    return 'search';
+  }
+
+  events() {
+    return {
+      'click .js-searchBtn': 'clickSearchBtn',
+      'change .js-sortBy': 'changeSortBy',
+      'change .js-filterWrapper select': 'changeFilter',
+      'change .js-filterWrapper input': 'changeFilter',
+      'keyup .js-searchInput': 'onKeyupSearchInput',
+    };
+  }
+
+  get sortByQuery() {
+    // return current sortBy state in the form of a query string
+    return this.$sortBy && this.$sortBy.length ? `&sortBy=${this.$sortBy.val()}` : '';
+  }
+
+  get filterQuery() {
+    // return all currently active filters in the form of a query string
+    return this.$filters && this.$filters.length ? `&${this.$filters.serialize()}` : '';
+  }
+
+  processTerm(term) {
+    this.term = term || '';
+    // if term has spaces, replace them with +
+    // if term is false, search for *
+    const query = `q=${term ? term.replace(/\s+/g, '+') : '*'}`;
+    const page = `&p=${this.serverPage}&ps=${this.pageSize}`;
+    const searchURL = `${this.sProvider}?${query}${this.sortByQuery}${this.filterQuery}${page}`;
+
+    this.callSearchProvider(searchURL);
+  }
+
+  callSearchProvider(searchURL) {
+    // remove a pending search if it exists
+    if (this.callSearch) this.callSearch.abort();
+
+    // initial render to show the loading spinner
+    this.render();
+
+    // query the search provider
+    this.callSearch = $.get({
+          url: searchURL,
+          dataType: 'json',
+        })
+        .done((data, status, xhr) => {
+        // make sure minimal data is present
+        if (data.name && data.links) {
+      this.render(data, searchURL);
+    } else {
+      this.showSearchError(xhr);
+      this.render({}, searchURL);
+    }
+  })
+  .fail((xhr) => {
+      if (xhr.statusText !== 'abort') {
+      this.showSearchError(xhr);
+      this.render({}, searchURL);
+    }
+  });
+  }
+
+  showSearchError(xhr = {}) {
+    const title = app.polyglot.t('search.errors.searchFailTitle', { provider: this.sProvider });
+    const failReason = xhr.responseJSON ? xhr.responseJSON.reason : '';
+    const msg = failReason ?
+                app.polyglot.t('search.errors.searchFailReason', { error: failReason }) : '';
+    const buttons = [];
+    if (this.usingDefault) {
+      buttons.push({
+        text: app.polyglot.t('search.changeProvider'),
+        fragment: 'changeProvider',
+      });
+    } else {
+      buttons.push({
+        text: app.polyglot.t('search.useDefault',
+          { term: this.term, defaultProvider: app.localSettings.get('searchProvider') }),
+        fragment: 'useDefault',
+      });
+    }
+
+    const errorDialog = new Dialog({
+      title,
+      msg,
+      buttons,
+      showCloseButton: false,
+    }).render().open();
+    this.listenTo(errorDialog, 'click-changeProvider', () => {
+      this.changeProvider();
+    errorDialog.close();
```

Comment:
Looks like some indentation is off here and also the `errorDialog.close` a few lines down.